File: pet-store-service/src/main/java/com/chewy/osn/factory/RepositoryFactory.java
package com.chewy.osn.factory;

import com.chewy.osn.config.RepositoryConfig;
import com.chewy.osn.repository.PetRepository;
import com.chewy.osn.repository.SchemaDefinition;
import io.micronaut.context.annotation.Bean;
import io.micronaut.context.annotation.Factory;
import jakarta.inject.Singleton;

@Factory
public class RepositoryFactory {

    @Bean
    @Singleton
    public PetRepository getPetRepository(RepositoryConfig config) {
        return new PetRepository(config, SchemaDefinition.PET_TABLE_SCHEMA);
    }
}

File: pet-store-service/src/main/kotlin/com/johnkruss/osn/factory/RepositoryFactory.kt
package com.johnkruss.osn.factory

import com.johnkruss.osn.config.RepositoryConfig
import com.johnkruss.osn.dynamo.PET_TABLE_SCHEMA
import com.johnkruss.osn.dynamo.PetRepository
import io.micronaut.context.annotation.Bean
import io.micronaut.context.annotation.Factory
import jakarta.inject.Singleton

@Factory
class RepositoryFactory {
    @Bean
    @Singleton
    fun getPetRepository(config: RepositoryConfig): PetRepository {
        return PetRepository(config, PET_TABLE_SCHEMA)
    }
}
